lib/FrontendTool/FrontendTool.cpp

```diff
@@ -685,6 +685,22 @@ static SourceFile *getPrimaryOrMainSourceFile(CompilerInvocation &Invocation,
   return SF;
 }
 
+/// Dumps the AST of all available primary source files. If corresponding output
+/// files were specified, use them; otherwise, dump the AST to stderr.
+static void dumpAST(CompilerInstance &Instance) {
+  for(SourceFile *sourceFile: Instance.getPrimarySourceFiles()) {
+    const StringRef OutputFilename =
+      Instance.getPrimarySpecificPathsForSourceFile(*sourceFile).OutputFilename;
+    // PSP.OutputFilename is "-" whenever the output should be written to the console.
+    if (OutputFilename == "-") {
```

Comment:
I think `getFileOutputStream` handles this, actually, by using `llvm::raw_fd_ostream`'s constructor. So you can drop this special case.